src/init.cc

```diff
@@ -1052,11 +1053,26 @@ static ncclResult_t initTransportsRank(struct ncclComm* comm, struct ncclComm* p
   NCCLCHECKGOTO(fillInfo(comm, comm->peerInfo+rank, comm->commHash), ret, fail);
   NCCLCHECKGOTO(bootstrapAllGather(comm->bootstrap, comm->peerInfo, sizeof(struct ncclPeerInfo)), ret, fail);
 
+  comm->busIdIdx = 0;
   for (int i = 0; i < nranks; i++) {
-    if ((i != rank) && (comm->peerInfo[i].hostHash == comm->peerInfo[rank].hostHash) && (comm->peerInfo[i].busId == comm->peerInfo[rank].busId)) {
-      WARN("Duplicate GPU detected : rank %d and rank %d both on CUDA device %lx", rank, i, comm->peerInfo[rank].busId);
-      ret = ncclInvalidUsage;
-      goto fail;
+    if ((i != rank) && (comm->peerInfo[i].hostHash == comm->peerInfo[rank].hostHash)) {
```

Comment:
correct place to calculate target GPUs should be at end of ncclTransportP2pSetup, because connections are established only during 1st collective call for sendrecv/alltoall type of collectives. comm->connectSend is the data structure to find target GPuUs info. It has multiple entries. We should only need to care about index 0 (all reduce type) and index 1 (sendrecv/alltoall type)